test/karma/test-app/prerender-test/karma.spec.ts

```diff
@@ -1,10 +1,30 @@
 import { setupDomTests, waitForChanges } from '../util';
 
+const orgConsoleError = window.console.error;
+
+/**
+ * This monkey-patches `window.console.error` in order to fail a test if that
+ * function is called within a test.
+ */
+function patchConsoleToFailOnError() {
+  window.console.error = function (...args: any[]) {
```

Comment:
Do we ever restore `window.console.error` anywhere after these tests run? Is that done implicitly by Jest in this scenario somehow (I don't recall if I'm honest).

Reply:
probs we should...I'll add some code for that

Reply:
I refactored the mocking function a little bit to be test-specific by doing the mock and then returning a teardown function to undo it, that way I can make the 'fail on console.error' behavior specific to just the one test I'm adding.